<launch>
  <!-- kinect and frame ids -->

  <!--
  <arg name="debug" default="false" />
  <arg if="$( arg debug )" name="launch_prefix" value="" />
  <arg unless="$( arg debug )" name="launch_prefix" value="" />
  <node pkg="openni_camera" type="openni_node" name="openni_node1" output="screen" launch-prefix="$( arg launch_prefix )">
    <param name="device_id" value="#1" />
    <rosparam command="load" file="$( find openni_camera )/info/openni_params.yaml" />
    <param name="rgb_frame_id" value="/openni_rgb_optical_frame" />
    <param name="depth_frame_id" value="/openni_depth_optical_frame" />
    <param name="use_indices" value="false" />
    <param name="depth_registration" value="false" />
    <param name="image_mode" value="8" />
    <param name="depth_mode" value="8" />
    <param name="debayering" value="0" />
    <param name="depth_time_offset" value="0" />
    <param name="image_time_offset" value="0" />
  </node>
  -->

  <include file="$( find openni_camera )/launch/kinect_frames.launch" />

  <node name="kinect_aux_functions" pkg="kinect_aux" type="kinect_aux_node">
    <remap from="tilt_angle" to="/kinect/angle" />
    <remap from="cur_tilt_angle" to="/kinect/cur_angle" />
    <remap from="imu" to="/kinect/accelerometer" />
  </node>

  <!-- openni manager -->
  <node pkg="nodelet" type="nodelet" name="openni_manager" output="screen" respawn="true" args="manager"/>

  <node pkg="nodelet" type="nodelet" name="openni_camera" args="load openni_camera/OpenNINodelet openni_manager" respawn="true">
    <param name="device_id" value="#1" />
    <rosparam command="load" file="$( find openni_camera )/info/openni_params.yaml" />
    <param name="rgb_frame_id" value="/openni_rgb_optical_frame" />
    <param name="depth_frame_id" value="/openni_depth_optical_frame" />
    <param name="use_indices" value="false" />
    <param name="depth_registration" value="false" />
    <param name="image_mode" value="8" />
    <param name="depth_mode" value="8" />
    <param name="debayering" value="0" />
    <param name="depth_time_offset" value="-0.055" />
    <param name="image_time_offset" value="0" />

    <rosparam command="load" file="$(find openni_camera)/info/openni_params.yaml" />
  </node>


  <!-- throttling -->
  <node pkg="nodelet" type="nodelet" name="pointcloud_throttle" args="load pcl_to_laser/CloudThrottle openni_manager">
    <param name="max_rate" value="2"/>
    <remap from="cloud_in" to="/camera/depth/points"/>
    <remap from="cloud_out" to="cloud_throttled"/>
  </node>

  <!-- fake laser -->
  <!--
  <node pkg="nodelet" type="nodelet" name="kinect_laser" args="load pcl_to_laser/CloudToScan openni_manager">
    <param name="output_frame_id" value="/openni_depth_frame"/>
    <remap from="cloud" to="cloud_throttled"/>
  </node>
  -->

  <!-- horizontal laser -->
  <node pkg="nodelet" type="nodelet" name="tilt_laser" args="load pcl_to_laser/CloudToScanHoriz openni_manager">
    <remap from="cloud_in" to="cloud_throttled"/>
    <param name="base_frame" value="/base_footprint"/>
    <param name="laser_frame" value="/base_footprint"/>
    <param name="min_height" value="-0.2"/>
    <param name="max_height" value="0.50"/>
  </node>

</launch>
